packages/vite/src/node/plugins/resolve.ts

```diff
@@ -465,20 +479,57 @@ export function tryNodeResolve(
 
 export function tryOptimizedResolve(
   id: string,
-  server: ViteDevServer
+  server: ViteDevServer,
+  importer?: string
 ): string | undefined {
   const cacheDir = server.config.cacheDir
   const depData = server._optimizeDepsMetadata
   if (cacheDir && depData) {
     const isOptimized = depData.optimized[id]
-    if (isOptimized) {
+
+    const getOptimizedUrl = (
+      optimizedEntry: typeof depData.optimized[string]
+    ) => {
       return (
-        isOptimized.file +
+        optimizedEntry.file +
         `?v=${depData.browserHash}${
-          isOptimized.needsInterop ? `&es-interop` : ``
+          optimizedEntry.needsInterop ? `&es-interop` : ``
         }`
       )
     }
+
+    // check if id has been optimized
+    if (isOptimized) {
+      return getOptimizedUrl(isOptimized)
+    }
+
+    // if has importer, further check if id is importing from nested depedency
+    if (importer) {
+      let resolvedSrc: string | undefined
+      for (const [k, v] of Object.entries(depData.optimized)) {
```

Comment:
**suggestion (dx):** Can we find better variable names as `k` and `v` here?

Reply:
Thanks for the review! Added better names for them and flatten the `tryOptimizedResolve()` function a bit to make it more readable.

Reply:
Much better now, thx